packages/table/src/cell/formats/truncatedFormat.tsx

```diff
@@ -178,34 +215,18 @@ export class TruncatedFormat extends React.Component<ITruncatedFormatProps, ITru
         }
 
         if (this.contentDiv === undefined) {
-            this.setState({ isTruncated: false });
+            this.setState({ isTruncated: false } as ITruncatedFormatState);
             return;
         }
 
-        const { isTruncated } = this.state;
-
-        // take all measurements at once to avoid excessive DOM reflows.
-        const {
-            clientHeight: containerHeight,
-            clientWidth: containerWidth,
-            scrollHeight: actualContentHeight,
-            scrollWidth: contentWidth,
-        } = this.contentDiv;
-
-        // if the content is truncated, then a popover handle will be present as a
-        // sibling of the content. we don't want to consider that handle when
-        // calculating the width of the actual content, so subtract it.
-        const actualContentWidth = isTruncated ? contentWidth - CONTENT_DIV_WIDTH_DELTA : contentWidth;
-
-        // we of course truncate the content if it doesn't fit in the container. but we
-        // also aggressively truncate if they're the same size with truncation enabled;
-        // this addresses browser-crashing stack-overflow bugs at various zoom levels.
-        // (see: https://github.com/palantir/blueprint/pull/1519)
-        const shouldTruncate =
-            (isTruncated && actualContentWidth === containerWidth) ||
-            actualContentWidth > containerWidth ||
-            actualContentHeight > containerHeight;
-
-        this.setState({ isTruncated: shouldTruncate });
+        if (this.cachedFontString == null) {
+            this.cachedFontString = Utils.getFontStringFromDom(this.contentDiv);
+        }
+
+        const contentWidth = Utils.measureText(this.contentDiv.textContent, this.cachedFontString).width;
+        const containerWidth = parseInt(this.props.parentCellWidth, 10);
+        const availableWidth = containerWidth - CONTAINER_PADDING;
+        const isTruncated = contentWidth > availableWidth;
+        this.setState({ isTruncated } as ITruncatedFormatState);
```

Comment:
// appears in a puff of smoke, surrounded by chanting and candles

Reply:
...um. Where'd height go? Why are we only measuring the width of the text itself? This doesn't actually work for any cases where we've got work wrap turned on. Did you test with that?

Reply:
so, uh, the only thing you can measure with canvas is width because that's all that exists on [TextMetrics](https://developer.mozilla.org/en-US/docs/Web/API/TextMetrics).

the text measurement assumes no wrapping in the text and is only useful at detecting horizontal overflow.

Reply:
![image](https://user-images.githubusercontent.com/569364/30933250-e631f4e2-a37e-11e7-908f-be2558dd446c.png)

😢